core/src/test/java/com/linecorp/armeria/internal/common/util/StringUtilTest.java

```diff
@@ -32,4 +32,37 @@ void intToString() {
         assertThat(StringUtil.toString(-1001)).isEqualTo("-1001");
         assertThat(StringUtil.toString(1001)).isEqualTo("1001");
     }
+
+    @Test
+    void testToBoolean() {
+        // Test supported boolean values
+        assertThat(StringUtil.toBoolean("true")).isTrue();
+        assertThat(StringUtil.toBoolean("TRUE")).isTrue();
+        assertThat(StringUtil.toBoolean("True")).isTrue();
+        assertThat(StringUtil.toBoolean("1")).isTrue();
+
+        assertThat(StringUtil.toBoolean("false")).isFalse();
+        assertThat(StringUtil.toBoolean("FALSE")).isFalse();
+        assertThat(StringUtil.toBoolean("False")).isFalse();
+        assertThat(StringUtil.toBoolean("0")).isFalse();
+    }
+
+    @Test
```

Comment:
nit

You can use `@parameterized` for such test cases: https://www.baeldung.com/parameterized-tests-junit-5